pallets/lbp/src/lib.rs

```diff
@@ -731,145 +761,82 @@ impl<T: Config> Pallet<T> {
-
-		if trade_type == TradeType::Sell {
-			ensure!(
-				T::MultiCurrency::free_balance(assets.asset_in, who) >= amount,
-				Error::<T>::InsufficientAssetBalance
-			);
-		}
-
-		let pool_id = Self::get_pair_id(assets);
-		let pool_data = <PoolData<T>>::try_get(&pool_id).map_err(|_| Error::<T>::PoolNotFound)?;
-
-		ensure!(Self::is_pool_running(&pool_data), Error::<T>::SaleIsNotRunning);
-
-		let now = T::BlockNumberProvider::current_block_number();
-
-		let (weight_in, weight_out) = Self::get_sorted_weight(assets.asset_in, now, &pool_data)?;
-
-		let asset_in_reserve = T::MultiCurrency::free_balance(assets.asset_in, &pool_id);
-		let asset_out_reserve = T::MultiCurrency::free_balance(assets.asset_out, &pool_id);
-
-		let (amount_in, amount_out, fee_asset, fee_amount) = if trade_type == TradeType::Sell {
-			ensure!(
-				amount <= asset_in_reserve.checked_div(MAX_IN_RATIO).ok_or(Error::<T>::Overflow)?,
-				Error::<T>::MaxInRatioExceeded
-			);
-
-			let token_amount_out = hydra_dx_math::lbp::calculate_out_given_in(
-				asset_in_reserve,
-				asset_out_reserve,
-				weight_in,
-				weight_out,
-				amount,
-			)
-			.map_err(|_| Error::<T>::Overflow)?;
-
-			let transfer_fee = Self::calculate_fees(&pool_data, token_amount_out)?;
-			let amount_out_without_fee = token_amount_out.checked_sub(transfer_fee).ok_or(Error::<T>::Overflow)?;
-
-			ensure!(limit <= amount_out_without_fee, Error::<T>::AssetBalanceLimitExceeded);
-			(amount, amount_out_without_fee, assets.asset_out, transfer_fee)
-		} else {
-			ensure!(
-				amount
-					<= asset_out_reserve
-						.checked_div(MAX_OUT_RATIO)
-						.ok_or(Error::<T>::Overflow)?,
-				Error::<T>::MaxOutRatioExceeded
-			);
-
-			let token_amount_in = hydra_dx_math::lbp::calculate_in_given_out(
-				asset_in_reserve,
-				asset_out_reserve,
-				weight_in,
-				weight_out,
-				amount,
-			)
-			.map_err(|_| Error::<T>::Overflow)?;
-
-			let transfer_fee = Self::calculate_fees(&pool_data, token_amount_in)?;
-			let amount_in_with_fee = token_amount_in.checked_add(transfer_fee).ok_or(Error::<T>::Overflow)?;
-
-			ensure!(limit >= amount_in_with_fee, Error::<T>::AssetBalanceLimitExceeded);
-			(token_amount_in, amount, assets.asset_in, transfer_fee)
-		};
-
-		if trade_type == TradeType::Buy {
-			ensure!(
-				T::MultiCurrency::free_balance(assets.asset_in, who) >= amount_in,
-				Error::<T>::InsufficientAssetBalance
-			);
+		match pool_data.start {
+			Some(start) => start <= now,
+			_ => false,
 		}
+	}
 
-		ensure!(
-			T::MultiCurrency::free_balance(assets.asset_out, &pool_id) >= amount_out,
-			Error::<T>::InsufficientAssetBalance
-		);
+	/// returns fees collected and locked in the fee collector account
+	/// note: after LBP finishes and liquidity is removed this will be 0
+	fn collected_fees(pool: &Pool<T::AccountId, T::BlockNumber>) -> BalanceOf<T> {
+		T::LockedBalance::get_by_lock(COLLECTOR_LOCK_ID, pool.assets.0, pool.fee_collector.clone())
```

Comment:
If you create another pool using the same account and collecting asset, collected fees from both LBPs are added up.